### PR TITLE
Bug 1238870 - [TV][2.5]Listen to Backspace not esc key for back function

### PR DESCRIPTION
**Pull request by rickychien**
https://bugzilla.mozilla.org/show_bug.cgi?id=1238870

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=610b48dbd2b4e8a33c301558dc2cace51e1ebe63

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5f41d34e1543fc58436c3c911aa1d784cf0a6a6a

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b578673197d84b50a355c4f4a33a703d409428cd

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6eb44ea2faa980ea5d2984550081db14a04869ff

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9677f245dfb9db096ce145a8dab35d69f31ac868

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5e7429baf29447e8e9ad61654c1b2209974229d0

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3cb8283873725d7698fec167ab230a9b002b9321

File: tv_apps/smart-system/js/keyboard_manager.js [hunk 1]
```diff
@@ -403,6 +403,10 @@ var KeyboardManager = {
     this.transitionManager.hideImmediately();
   },
 
+  getHasActiveKeyboard: function km_getHasActiveKeyboard() {
```
Comment (luke-chang) [inline tv_apps/smart-system/js/keyboard_manager.js]: I would suggest using the naming: `hasActiveKeyboard`.

File: tv_apps/smart-system/js/hardware_buttons.js [hunk 2]
```diff
@@ -229,11 +231,22 @@
...
       }
     } else {
       var buttonEventType = this.browserKeyEventManager.getButtonEventType(evt);
+
+      // In case of website only calling preventDefault() for keydown or keyup.
+      // If preventDefault() is called in keydown, we set to base state to exit
+      // previous state and force FSM do nothing for upcoming keyup / *-release.
+      // If preventDefault() is called in keyup, we set to base state to exit
+      // previous state for preventing "hold" event or infinity repeat
+      if (this.browserKeyEventManager.isAppCancelledKeyEvent(evt)) {
```
Comment (luke-chang) [inline tv_apps/smart-system/js/hardware_buttons.js]: check `embeddedCancelled` here.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9896750829d216534e830a41212e285d99d53493

**Pull request merged**